examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/main.c

```diff
@@ -589,20 +596,50 @@ int main( void )
     wifi_bool SSID_found = WIFI_FALSE;
     WiFi_Status_t status = WiFi_MODULE_SUCCESS;
 
+    /* Initialize uC and peripherals */
     if ( 0 > system_init() )
     {
         printf( "\r\n>> System initialization [ERROR] - System boot aborted" );
         while ( 1 )
             ;
     }
 
-    if ( 0 > user_config_init( &user_config ) )
+#if USE_HARDCODED_CREDENTIALS
+    user_data_set_wifi_ssid( (&user_config), USER_CONFIG_WIFI_SSID );
+    user_data_set_wifi_psk( (&user_config), USER_CONFIG_WIFI_PWD );
+    user_data_set_wifi_encryption( (&user_config), USER_CONFIG_WIFI_ENCR );
+    user_data_set_xi_account_id( (&user_config), USER_CONFIG_XI_ACCOUNT_ID );
+    user_data_set_xi_device_id( (&user_config), USER_CONFIG_XI_DEVICE_ID );
+    user_data_set_xi_device_password( (&user_config), USER_CONFIG_XI_DEVICE_PWD );
+#else
+    /* Read user data from flash */
+    if ( user_data_copy_from_flash( &user_config ) < 0 )
     {
-        printf( "\r\n>> User data initialization [ERROR] - System boot aborted" );
+        printf( "\r\n>> [ERROR] trying to copy user data from flash. Abort" );
         while ( 1 )
             ;
     }
+    printf( "\r\n>> User data retrieved from flash:" );
+    user_data_printf( &user_config );
+    fflush( stdout );
+
+    /* Provision the device if requested or if flash data is missing||corrupt */
+    int8_t provisioning_bootmode = io_read_button();
+    if ( provisioning_bootmode || ( 0 > user_data_validate_checksum( &user_config ) ) )
+    {
+        ( provisioning_bootmode == 1 )

```

Comment:
here the `provisioning_bootmode` is used in two different ways as a boolean value. First is just testing the value the second checks its equality with 1. Two things here:
- 1st: select one type of testing, I'd prefer checking the equality to 1 since - as I know - it is up to the compiler? how it represents boolean TRUE and FALSE values. I it isn't necessary that 1 is TRUE and 0 is FALSE - although I haven't seen any other solution yet.
- 2nd: in 9thBit we agreed upon to put the tested value at front to eliminate the possibility of accidental assignment which is hardly debuggable, so I'd advise to use there `1 == provisioning_bootmode` at both places :)